### ISSUE TITLE
Offcanvas Rendering Incorrectly on Chrome Mobile View

### ISSUE THREAD

**Issue by Dave-LeDev** [gantry/gantry5#1224]
Demonstration can be found here: http://daveledev.com/joomla/

Instead of getting the 3-bar button, it renders the menu as if it were in desktop view. If I have the site on a full window on desktop, the error doesn't throw until I scale the window down to mobile dimensions or simulate a mobile view. So something's triggering for mobile view and it gets caught when it gets to an iframe with a domain or protocol mismatch.

**Simulated Result**
![image](https://cloud.githubusercontent.com/assets/18293902/14309146/2c5cd870-fba0-11e5-93ee-ddc74303b51f.png)

**Intended Result**
![image](https://cloud.githubusercontent.com/assets/18293902/14309131/1609c902-fba0-11e5-9313-20498b951bec.png)

Simply embed a video. I grabbed the code straight from YouTube. My main menu has multiple items. It's easiest to see this error on a mobile device or simulating a mobile device in Chrome. It's not guaranteed to throw the error if you start with a desktop layout, depending on the order JS is executed.

Tested browsers:
Android 5.1 Chrome
Windows 10x64 Chrome 49.0.2623.110 m

**Comment by mahagr** [comment]
The error doesn't happen every time, in fact I wasn't able to get it at all in my phone.

**Comment by alisamii** [comment]
It doesn't happen on safari or chrome on iPhone.

**Comment by JoomFX** [comment]
I wasn't able to reproduce it in:

Desktop (simulating phone view), Win 8.1, Firefox 45.0.1
Desktop (simulating phone view), Win 8.1, Chrome 49.0.2623.110
HTC One, Android 5.0.2, Chrome browser

**Comment by Dave-LeDev** [comment]
With earlier provided demo link, my steps to reproduce:

1. Open Chrome inspector
2. Toggle Device Mode (Ctrl + Shift + M)
3. Mine is set to Galaxy S5 but it should work on any other mobile simulation
4. Load http://daveledev.com/joomla/
5. Refresh as needed to recreate; should do it after the first or second refresh

I've reproduced this issue on a Moto X 2013. I'll try another mobile device when I can get my hands on one.

**Comment by Dave-LeDev** [comment]
Another user reported this issue, so I know I'm not alone.

https://files.gitter.im/gantry/gantry5/eCOV/thumb/Image.png

**Comment by Dave-LeDev** [comment]
Weird. Now I can't get the bug to trigger on my phone. I wonder if Gantry also detects the browser and so is only isolated to Windows. When I can I'll try it on another desktop.

**Comment by Dave-LeDev** [comment]
Either that or I had some cached junk code and it finally cleared. Still doesn't explain the bug on desktop, though.

**Comment by Dave-LeDev** [comment]
FOUND IT!! The culprit is the [ExtraWatch](https://www.extrawatch.com/) Agent Module. Upon disabling, Offcanvas loads properly, and the error is never thrown. No one else could see this because that module is visible only to Super Admin. Strange how it mucks with Gantry, though.

**Comment by Dave-LeDev** [comment]
Eeeh~ Premature celebration. Still doing it. -_-

**Comment by Dave-LeDev** [comment]
http://gadunited.com/index.php?option=com_easyblog&view=latest&Itemid=465

Try this one out. This is the only one I can't seem to correct out of shear luck.

**Comment by mahagr** [comment]
I cannot still get it to happen, ever. I hope you have a better luck to tracking it down.

**Comment by Dave-LeDev** [comment]
I don't get it, though. I'm able to replicate on 2 Windows machines, a Macbook, and 2 Android devices running Lollipop and Marshmallow. And two other users were able to replicate it too. Ugh!  Such a frustrating problem!!

**Comment by Dave-LeDev** [comment]
Would you be able to point me in the direction of what's happening with this screenshot?
https://i.gyazo.com/03a4deb893a13068ed176c9111b025eb.png

**Comment by JoomFX** [comment]
@Dave-LeDev I just saw [your post](https://www.facebook.com/groups/joomlanospam/?multi_permalinks=10153374700145997) in the Joomla Group on Facebook.
As you can see, the guys there are also not able to see the issue you talk about.

Gantry 5 is the best (oh boy the best by far) framework you can find on the market. And, as you already know, it is FREE.

If there is an issue, a bug in Gantry 5, it will be solved. You can be sure in that. We just need to be able to reproduce it. How could you fix something you cannot reproduce?

The guys here work their a** off so we can all have this awesome framework, so you might want to reconsider your opinion about the Gitter chat being dead and the quality of support you have received (for a FREE product). Once again, if the issue is in Gantry 5, it will be fixed, you can count on that. Just find a consistent way to reproduce the issue (on more than one computer so you know that your environment is not the root cause) and provide the exact steps.

**Comment by Dave-LeDev** [comment]
Whoa there, there seems to be a misunderstanding of dialect. I'm grateful for the free support and I'm frustrated by the problem itself, not of you guys. In regards to the "chat being dead", that was to describe the activity that was going on when I joined in. There was no communication for hours. I've even been looking for some kind of support that I can pay for so someone can help look into my website.

**Comment by mahagr** [comment]
Regarding to gitter chat, its meaning is not to be a support channel. Its just a chat where you can talk with other users and developers. Most projects do not offer you to do that and it has been proven quite useful resource if you have the patience and ask the right questions.

**Comment by Dave-LeDev** [comment]
Ah, I was mis-directed on its purpose. Sincerest apologies.

**Comment by w00fz** [comment]
If you look at the error, the issue is that youtube refuses to load in your site because of the different protocol http vs https.

> main.js?5704b739:1 Uncaught SecurityError: Blocked a frame with origin "http://daveledev.com" from accessing a frame with origin "https://www.youtube.com".  The frame requesting access has a protocol of "http", the frame being accessed has a protocol of "https". Protocols must match.

Instead of using `https://youtube.com/....` try using `http://youtube.com...`

**Comment by mahagr** [comment]
Or just use `//www.youtube.com`, it should work too.

**Comment by w00fz** [comment]
Well when I posted my answer above I admit I didn't first check the  link @JoomFX referred to, now I did and I can see you received help where others are practically telling you exactly the same as above. The issue is not Gantry but YouTube and the fact that you have no SSL. Try one of the two solutions provided in the :point_up: comments, I think they might help (and I'm also curious to know if it does!).

Either way I'm going to close this issue as it's clearly not Gantry related. Please feel free to reopen if you have any additional details / informations to provide.

Thanks!

**Comment by w00fz** [comment]
Actually @Dave-LeDev, I might have a workaround in Gantry to circumvent the error that youtube throws (or whatever remote iframe loaded for that matter).

Care to try with the latest [CI Build](http://gantry.org/downloads#ci-builds)? We are planning a release quite soon so it would be nice if you could test and let me know asap!

Thanks!

**Comment by Dave-LeDev** [comment]
I did find a solution to the problem, and it's quite embarrassing. I should have done this from the beginning. Tried Microsoft Edge browser for giggles.

THIS SON OF A &!7{# IS WHAT'S BREAKING IT

https://chrome.google.com/webstore/detail/stylebot/oiaejidbmkiecgbjeifoejpgmdaleoha?utm_source=chrome-app-launcher-info-dialog

Moral of the story? Keep troubleshooting simple and watch out for seemingly useful browser tools.
